codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/s3/PresignPostCustomization.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.go.codegen.customization.s3;
+
+import software.amazon.smithy.go.codegen.GoCodegenContext;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+
+import java.util.Map;
+import java.util.Optional;
+
+import static software.amazon.smithy.go.codegen.GoWriter.goTemplate;
+
+/**
+ *  CodeGen class for a presigner that uses POST
+ *  This is a special kind of presigner that instead of generating and signing an HTTP request,
+ *  generates a URL and field values to be used on an HTML form.
+ * <p>
+ *  This method of sending request is only implemented service side for putting objects into an S3 bucket.
+ *  However, it could be expanded to other operations if other services start supporting it
+ *
+ *  @see <a href="https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html> the public docs</a>
+ */
+public class PresignPostCustomization implements GoIntegration {
```

Comment:
Is there a reason we need to codegen this? It could just live in its own file in s3 unless I'm missing something.

Reply:
I know other presigned APIs get generated into their operation's files but I'm not concerned about matching that convention for handwritten stuff like this. Always in favor of explicitly written Go code where we can instead of generated.

Reply:
Moved to written Go.

Originally I wasn't sure how much of this could be extended to other presigners, like whether we could this kind of signer for other operations. After some internal discussions, it doesn't seem like there are plans to extend this to other operations